From 7532c6838e41e65c8f87b08ebc1653dfa8a6ea42 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:13 -0400
Subject: [powerpc] legacy_serial: Fix incorrect placement of __initdata tag

Message-id: <1410545655-205645-85-git-send-email-dzickus@redhat.com>
Patchwork-id: 94586
O-Subject: [RHEL7 PATCH 084/626] powerpc/legacy_serial: Fix incorrect placement of __initdata tag
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit dc1473dcfaf4fb38ba2dcb2fb7ac7d0242185fa3
Author: Bartlomiej Zolnierkiewicz <b.zolnierkie@samsung.com>
Date:   Mon Sep 30 15:11:42 2013 +0200

    powerpc/legacy_serial: Fix incorrect placement of __initdata tag

    __initdata tag should be placed between the variable name and equal
    sign for the variable to be placed in the intended .init.data section.

    Signed-off-by: Bartlomiej Zolnierkiewicz <b.zolnierkie@samsung.com>
    Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/legacy_serial.c b/arch/powerpc/kernel/legacy_serial.c
index 22e88dd..40bd7bd 100644
--- a/arch/powerpc/kernel/legacy_serial.c
+++ b/arch/powerpc/kernel/legacy_serial.c
@@ -35,7 +35,7 @@ static struct legacy_serial_info {
  phys_addr_t   taddr;
 } legacy_serial_infos[MAX_LEGACY_SERIAL_PORTS];
 
-static struct __initdata of_device_id legacy_serial_parents[] = {
+static struct of_device_id legacy_serial_parents[] __initdata = {
  {.type = "soc",},
  {.type = "tsi-bridge",},
  {.type = "opb", },
-- 
1.7.1